### PR TITLE
Optional CI jobs for containerd/containerd repo

### PR DESCRIPTION
**Pull request by dims**
Blindly clone containerd/cri CI jobs, but
- run only on master
- optional
- run only on demand

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

### CODE REVIEW

**Comment by dims** [comment]
cc @MHBauer @mikebrow

File: config/jobs/containerd/containerd/containerd-presubmit-jobs.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      description: build artifacts
+    labels:
+      preset-service-account: "true"
+    spec:
+      containers:
+      - image: gcr.io/k8s-testimages/kubekins-e2e:v20200901-eeeadc5-master
+        command:
+        - runner.sh
+        args:
+        - ./test/build.sh
```
Comment (mikebrow) [inline config/jobs/containerd/containerd/containerd-presubmit-jobs.yaml:21]: will need to add a script in containerd/containerd or move to make..

File: config/jobs/containerd/containerd/containerd-presubmit-jobs.yaml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    spec:
+      containers:
+      - name: pull-containerd-node-e2e
+        image: gcr.io/k8s-testimages/kubekins-e2e:v20200901-eeeadc5-master
+        command:
+        - sh
+        - -c
+        - >
+          runner.sh
+          ./test/build.sh
```
Comment (mikebrow) [inline config/jobs/containerd/containerd/containerd-presubmit-jobs.yaml:57]: ..

File: config/jobs/containerd/containerd/containerd-presubmit-jobs.yaml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    spec:
+      containers:
+      - image: gcr.io/k8s-testimages/kubekins-e2e:v20200901-eeeadc5-master
+        args:
+        - "--repo=github.com/containerd/$(REPO_NAME)=$(PULL_REFS)"
+        - "--root=/go/src"
+        - "--upload=gs://kubernetes-jenkins/pr-logs"
+        - --scenario=execute
+        - --
+        - ./test/verify.sh
```
Comment (mikebrow) [inline config/jobs/containerd/containerd/containerd-presubmit-jobs.yaml:92]: ..

File: config/jobs/containerd/containerd/containerd-presubmit-jobs.yaml [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+          /workspace/scenarios/kubernetes_e2e.py
+          --deployment=node
+          --gcp-project=cri-c8d-pr-node-e2e
+          --gcp-zone=us-central1-f
+          '--node-test-args=--container-runtime=remote --container-runtime-endpoint=unix:///run/containerd/containerd.sock --container-runtime-process-name=/home/containerd/usr/local/bin/containerd --container-runtime-pid-file= --kubelet-flags="--cgroups-per-qos=true --cgroup-root=/ --runtime-cgroups=/system.slice/containerd.service" --extra-log="{\"name\": \"containerd.log\", \"journalctl\": [\"-u\", \"containerd\"]}"'
+          --node-tests=true
+          --provider=gce
+          '--test_args=--nodes=8 --focus="\[NodeConformance\]|\[NodeFeature:.+\]" --skip="\[Flaky\]|\[Slow\]|\[Serial\]"'
+          --timeout=65m
+          "--node-args=--image-config-file=${GOPATH}/src/k8s.io/test-infra/jobs/e2e_node/containerd/cri-master/image-config-pr.yaml -node-env=PULL_REFS=$(PULL_REFS)"
```
Comment (mikebrow) [inline config/jobs/containerd/containerd/containerd-presubmit-jobs.yaml:70]: have to replace cri-master with a new set of containerd-master files...

Comment (mikebrow) [reply config/jobs/containerd/containerd/containerd-presubmit-jobs.yaml:70]: which may have to reference new replacements for  containerd/cri/cluster/gce*

File: config/jobs/containerd/containerd/containerd-presubmit-jobs.yaml (hunk 1 above)
Comment (dims) [reply config/jobs/containerd/containerd/containerd-presubmit-jobs.yaml:21]: Correct. we can tweak this later.

File: config/jobs/containerd/containerd/containerd-presubmit-jobs.yaml (hunk 4 above)
Comment (dims) [reply config/jobs/containerd/containerd/containerd-presubmit-jobs.yaml:70]: This is from the `k8s.io/test-infra` repo. not `containerd/cri` repo

File: config/jobs/containerd/containerd/containerd-presubmit-jobs.yaml (hunk 3 above)
Comment (dims) [reply config/jobs/containerd/containerd/containerd-presubmit-jobs.yaml:92]: same as above. we can figure out if we need this and add a script or switch make targets

File: config/testgrids/kubernetes/sig-node/config.yaml [hunk 5]
```diff
@@ -17,6 +18,7 @@ dashboard_groups:
 
 
 dashboards:
+- name: sig-node-containerd-io
```
Comment (MHBauer) [inline config/testgrids/kubernetes/sig-node/config.yaml:21]: Ooh, I can't wait to see if this goes to first in the list.

I want to suggest putting it in the same order as the other list, but I don't know how the inside of testgrid works.

**Review by MHBauer — APPROVED** [review]
If it's a copy it's a copy, I'm not going to look to carefully at the job description.

**Pull request merged**